app/presenters/publishing_api/historical_accounts_index_presenter.rb

```diff
@@ -0,0 +1,69 @@
+module PublishingApi
+  class HistoricalAccountsIndexPresenter
+    attr_accessor :update_type
+
+    def initialize(update_type: nil)
+      self.update_type = update_type || "major"
+    end
+
+    def content_id
+      "a258e45a-acbe-4d70-ad2c-a2a20761536a"
+    end
+
+    def content
+      content = BaseItemPresenter.new(
+        nil,
+        title: "Past Prime Ministers",
+        update_type:,
+      ).base_attributes
+
+      content.merge!(
+        base_path:,
+        details: {
+          appointments_without_historical_accounts:,
+        },
+        document_type: "historic_appointments",
+        public_updated_at: Time.zone.now,
+        rendering_app: Whitehall::RenderingApp::WHITEHALL_FRONTEND,
+        schema_name: "historic_appointments",
+      )
+
+      content.merge!(PayloadBuilder::Routes.for(base_path))
+    end
+
+    def find_appointments_without_accounts(role)
+      past_pms = role.role_appointments
+      ids_of_pms_without_accounts = past_pms.map(&:person_id) - HistoricalAccount.all.map(&:person_id)
+      Person.where(id: ids_of_pms_without_accounts)
+    end
+
+    def appointments_without_historical_accounts
+      role = Role.friendly.find("prime-minister")
+      people_to_present = find_appointments_without_accounts(role)
+      people_to_present.map do |person|
+        previous_role_appointments = RoleAppointment.where(person_id: person.id)
+                                                    .where.not(RoleAppointment::CURRENT_CONDITION)
```

Comment:
I wonder if it's worth creating a `historic` scope of the `RoleAppointment` model (since we already have one called `current`, doing the inverse of this)?

Something like:
```
 scope :historic, -> { where.not(CURRENT_CONDITION) }
```

Then this could be updated to, then in-lined on line 49:
```
previous_role_appointments = person.role_appointments.historic
```

Reply:
Good suggestion. Also made me realise that I wasn't actually just presenting historic role appointments here, meaning that this errored if someone had both historic and current appointments. Amended the test to add this case in too